mods/pageviews.js

```diff
@@ -0,0 +1,217 @@
+    this.log = options.log || function() {};
+}
+
+
+var tables = {
+        article: 'pageviews.per.article',
+        project: 'pageviews.per.project',
+        tops: 'top.pageviews',
+    };
+var tableURI = function(domain, tableName) {
+        return new URI([domain, 'sys', 'table', tableName, '']);
+    };
+var tableSchemas = {
+        article: {
+            table: tables.article,
+            version: 1,
+            attributes: {
+                project: 'string',
+                article: 'string',
+                access: 'string',
+                agent: 'string',
+                granularity: 'string',
+                // the hourly timestamp will be stored as YYYYMMDDHH
+                timestamp: 'string',
+                views: 'int'
+            },
+            index: [
+                { attribute: 'project', type: 'hash' },
+                { attribute: 'article', type: 'hash' },
+                { attribute: 'access', type: 'hash' },
+                { attribute: 'agent', type: 'hash' },
+                { attribute: 'granularity', type: 'hash' },
+                { attribute: 'timestamp', type: 'range', order: 'asc' },
+            ]
+        },
+        project: {
+            table: tables.project,
+            version: 1,
+            attributes: {
+                project: 'string',
+                access: 'string',
+                agent: 'string',
+                granularity: 'string',
+                // the hourly timestamp will be stored as YYYYMMDDHH
+                timestamp: 'string',
+                views: 'int'
+            },
+            index: [
+                { attribute: 'project', type: 'hash' },
+                { attribute: 'access', type: 'hash' },
+                { attribute: 'agent', type: 'hash' },
+                { attribute: 'granularity', type: 'hash' },
+                { attribute: 'timestamp', type: 'range', order: 'asc' },
+            ]
+        },
+        tops: {
+            table: tables.tops,
+            version: 1,
+            attributes: {
+                project: 'string',
+                access: 'string',
+                year: 'string',
+                month: 'string',
+                day: 'string',
+                // format for this is a json array: [{rank: 1, article: <<title>>, views: 123}, ...]
+                articles: 'string'
+            },
+            index: [
+                { attribute: 'project', type: 'hash' },
+                { attribute: 'access', type: 'hash' },
+                { attribute: 'year', type: 'hash' },
+                { attribute: 'month', type: 'hash' },
+                { attribute: 'day', type: 'hash' },
+            ]
+        }
+    };
+
+/* general handler functions */
+var queryCatcher = function(e) {
+        if (e.status !== 404) {
+            throw e;
+        }
+    };
+var queryResponser = function(res) {
+        // always return at least an empty array so that queries for non-existing data don't error
```

Comment:
Yeah, I was torn about this as well.  I propose we leave it like this as an experiment and talk to people as they use the service.  Sadly, there's no way to say "this data should exist but doesn't".  I leaned towards the empty result because I find that easier to deal with on the front end (in dashboards, etc.) in cases where I'm expecting data to be there but it's just not ready yet or something.